Apps/W1/Shopify/app/src/Customer/Codeunits/ShpfyCustByBillto.Codeunit.al

```diff
@@ -54,28 +54,28 @@ codeunit 30111 "Shpfy Cust. By Bill-to" implements "Shpfy ICustomer Mapping"
             ShopifyCustomer.Insert();
             CustomerApi.RetrieveShopifyCustomer(ShopifyCustomer);
         end;
-        Address.SetAutoCalcFields("Customer No.");
-        Address.SetRange("Customer Id", CustomerId);
-        Address.SetFilter("Address 1", FilterMgt.CleanFilterValue(JsonHelper.GetValueAsText(JCustomerInfo, 'Address'), MaxStrLen(Address."Address 1")));
-        Address.SetFilter("Address 2", FilterMgt.CleanFilterValue(JsonHelper.GetValueAsText(JCustomerInfo, 'Address2'), MaxStrLen(Address."Address 2")));
-        Address.SetRange(Zip, JsonHelper.GetValueAsText(JCustomerInfo, 'PostCode'));
-        Address.SetRange(City, JsonHelper.GetValueAsText(JCustomerInfo, 'City'));
-        Address.SetRange("Country/Region Code", JsonHelper.GetValueAsText(JCustomerInfo, 'CountryCode'));
-        if Address.IsEmpty then
+        CustomerAddress.SetAutoCalcFields("Customer No.");
+        CustomerAddress.SetRange("Customer Id", CustomerId);
+        CustomerAddress.SetFilter("Address 1", FilterMgt.CleanFilterValue(JsonHelper.GetValueAsText(JCustomerInfo, 'Address'), MaxStrLen(CustomerAddress."Address 1")));
```

Comment:
this is interesting. the clean does this:
        exit('@' + Value.Replace('(', '?').Replace(')', '?').Replace('*', '?').Replace('.', '?').Replace('<', '?').Replace('>', '?').Replace('=', '?'));


So basically make a case insensitive wildcard filter. But, this could get you wrong results. The ? will accept any character, not only the ones that mess with the filterstring. 

Also, there is a good way of filtering that works even when the value contains these chars and this is not it.

Say the address is 
'BC (AL)',

this filter thing will return anything that has 'BC ?AL?' (case insensitive). like, BC HALO but that's not correct is it? 

What I wonder about is whether there was actually an issue or its this way because you thought you had to avoid one. 
If the cases insensitive thing isn't truly required, a SetRange should have done the trick. 

it seems to me the loopy code below is to weed out all the addresses this finds that are not the ones you're looking for... 

I'd be interested in seeing an actual case where this is needed and then trying to find an easier solution to the filtering.